### PR TITLE
web app. primary languages 

### PR DESCRIPTION
**Pull request by brandelune**

### CODE REVIEW

File: questionnaire-de.json [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+  "numberColumnText": "Nummer",
+  "specRefColumnText": "Spec Ref",
+  "questionColumnText": "Fragentext",
+  "tocText": "Inhaltsverzeichnis",
+  "preambleText": "Das OpenChain-Projekt schafft Vertrauen in Open Source, indem es die Lizenzkonformität einfacher und konsistenter macht. Es verwaltet die OpenChain-Spezifikation, um einen Kernsatz von Anforderungen zu definieren, die jedes Qualitätskonformitätsprogramm erfüllen muss. Das Ergebnis ist, dass die Einhaltung von Open Source-Lizenzen für die Teilnehmer der Software-Lieferkette vorhersehbarer, verständlicher und effizienter wird.\nDieses Dokument enthält eine Reihe von Fragen, um festzustellen, ob ein Unternehmen über ein OpenChain-Konformitätsprogramm verfügt. Wenn jede dieser Fragen mit \"Ja\" beantwortet werden kann, erfüllt dieses Unternehmen alle Anforderungen an die Konformität mit der OpenChain-Spezifikation Version 2.1 (funktional identisch mit ISO \/ IEC DIS 5230: 2020 (e)). Wenn eine der Fragen mit \"Nein\" beantwortet wird, kann das Unternehmen klar erkennen, wo zusätzliche Investitionen erforderlich sind, um den Compliance-Prozess zu verbessern.",
+  "sections": [
+    {
+      "name": "Teil 1",
```
Comment (goneall) [inline questionnaire-de.json:18]: The section names should not be translated - they are metadata and not displayed

File: questionnaire-ja.json [hunk 2]
```diff
@@ -1,434 +1,434 @@
...
-          "correctAnswer": "Yes",
-          "question": "Do you have a process for determining the scope of your Program?",
-          "number": "1.g",
-          "type": "YES_NO",
+            "1.1",
+            "1.1.2"
+          ]
+        },
+        {
+          "correctAnswer": "はい",
```
Comment (goneall) [inline questionnaire-ja.json:42]: The correctAnswer should not be translated

File: questionnaire-ko.json [hunk 3]
```diff
@@ -1,434 +1,434 @@
...
+  "questionColumnText": "질문 내용",
+  "tocText": "목차",
+  "preambleText": "OpenChain 프로젝트는 오픈소스 라이선스 컴플라이언스를 보다 간단하고 일관성있게 만들어 오픈소스에 대한 신뢰를 구축합니다. OpenChain 설명서는 모든 우수한 컴플라이언스 프로그램이 충족해야하는 핵심 요건을 정의합니다. 그 결과 소프트웨어 공급망의 참가자가 오픈소스 라이선스 컴플라이언스를 보다 예측 가능하고 이해하기 쉽고 효율적으로 만들 수 있습니다.\n이 문서에는 회사가 OpenChain을 준수하는지 여부를 확인하기위한 일련의 질문이 포함되어 있습니다. 이러한 각 질문에 \"예\"라고 대답 할 수 있으면 해당 회사는 OpenChain 설명서 버전 2.0에 대한 모든 요건을 충족합니다. 질문에 \"아니오\"라고 대답하면 회사는 컴플라이언스 프로세스를 개선하기 위해 추가 투자가 필요한 위치를 명확하게 식별 할 수 있습니다.",
+  "sections": [
+    {
+      "name": "Part 1",
+      "title": "1. 프로그램 설립",
+      "questions": [
+        {
+          "correctAnswer": "네",
```
Comment (goneall) [inline questionnaire-ko.json:22]: The correctAnswer should not be translated

File: questionnaire-ko.json [hunk 4]
```diff
@@ -1,434 +1,434 @@
...
-          "question": "Is the external open source compliance contact publicly identified (e.g. via an email address or the Linux Foundation Open Compliance Directory)?",
-          "number": "2.b",
-          "type": "YES_NO",
+            "1.5.1"
+          ]
+        }
+      ]
+    },
+	{
+      "name": "Yes",
```
[the remark this answers is not in the document]
Comment (goneall) [reply questionnaire-ko.json:152]: Actually, section names should not be translated - just need to replace with "Part 2"

File: questionnaire-ko.json [hunk 5]
```diff
@@ -1,434 +1,434 @@
...
+          "question": "Part 2",
+          "number": "2.a",
+          "type": "YES_NO",
+          "specReference": [
+            "2.1",
+            "2.2.1"
+          ]
+        },
+        {
+          "correctAnswer": "네",
```
Comment (goneall) [inline questionnaire-ko.json:166]: correctAnswer should not be translated

File: questionnaire-ko.json [hunk 6]
```diff
@@ -1,434 +1,434 @@
...
-        {
-		  "question": "Do you have a documented procedure that covers these common open source license use cases for open source components in the Supplied Software?",
-          "number": "3.c",
-          "type": "SUBQUESTIONS",
+            "3.1",
+            "3.1.2"
+          ]
+        },
+        {
+		  "question": "Yes",
```
Comment (goneall) [inline questionnaire-ko.json:262]: This does not look correct - it should be a translation of the question 3.1.2

File: questionnaire-ko.json [hunk 7]
```diff
@@ -1,434 +1,434 @@
...
-          "type": "YES_NO",
+            	"3.2.1"
+              ]
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "Yes",
```
Comment (goneall) [inline questionnaire-ko.json:336]: "name" is incorrect - it should be "Part 4"

File: questionnaire-ko.json [hunk 8]
```diff
@@ -1,434 +1,434 @@
...
-          "number": "6.a",
-          "type": "YES_NO",
+            "5.1",
+            "5.1.3"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Yes",
```
Comment (goneall) [inline questionnaire-ko.json:408]: This is incorrect - it should be "Part 6"

File: questionnaire-ko.json [hunk 9]
```diff
@@ -1,434 +1,434 @@
...
+      "title": "6. 5.c",
```
Comment (goneall) [inline questionnaire-ko.json:409]: This is incorrect - it should be a translation of the title

File: questionnaire-ko.json [hunk 10]
```diff
@@ -1,434 +1,434 @@
...
+      "questions": [
+        {
+          "correctAnswer": "네",
+          "question": "Part 6",
```
Comment (goneall) [inline questionnaire-ko.json:413]: This is incorrect - it should be a translation of the question

File: questionnaire-zh-tw.json [hunk 11]
```diff
@@ -0,0 +1,434 @@
...
+  "questionsText": "問題",
+  "completedText": "已完成?",
+  "statusText": "狀態",
+  "answerColumnText": "回答",
+  "sectionColumnText": "部份",
+  "numberColumnText": "編號",
+  "specRefColumnText": "規格參考",
+  "questionColumnText": "問題集",
+  "tocText": "目錄",
+  "preambleText": "OpenChain工作項目通過簡化許可開源的合規性和保持一致性，建立起對於開放源的信任。為維護OpenChain規範 ，在定義核心數據方面必需満足合規程序的各項品質要求。其結果是，對於軟體供應鍵參與者而言，開源許可合規性變得更加可預測、可理解和高效率。\n本問卷包含一系列問題，以確認該公司是否符合OpenChain合格程序。如果所有問題的回答都為“是”，則該公司符合OpenChain規範版本2.1（功能與ISO\/IEC DIS 5230:2020（e）相同）的所有要求。如果任一問題的回答為“否”，則該公司可以明確的知道需要在哪些項目加以改善以達成合規程序。",
```
Comment (goneall) [inline questionnaire-zh-tw.json:15]: I'm not sure about the backslash in the above preamble text

File: questionnaire-zh-tw.json [hunk 12]
```diff
@@ -0,0 +1,434 @@
...
+  "sections": [
+    {
+      "name": "Part 1",
+      "title": "1. 基礎程序",
+      "questions": [
+        {
+          "correctAnswer": "是",
```
Comment (goneall) [inline questionnaire-zh-tw.json:22]: correctAnswer should not be translated

File: questionnaire-zh-zh.json [hunk 13]
```diff
@@ -0,0 +1,434 @@
...
+  "questionColumnText": "问题文本",
+  "tocText": "目录",
+  "preambleText": "OpenChain项目通过简化开放源许可合规性和保持一致性，建立对于开放源的信任。维护OpenChain规范，定义各品质合规程序必须满足的一组核心需求。其结果是，对于软件供应链参与者来说，开放源许可合规性变得更加可预测、可理解和高效。\n本文档包含一系列确定公司是否符合OpenChain的问题。如果所有问题的答案都为“是”，那么该公司符合OpenChain规范2.1版（功能与ISO\/IEC DIS 5230:2020（e）一致）的所有要求。如果任一问题的答案为“否”，那么公司可以清楚地确定哪些需要追加投资以改进合规程序。",
+  "sections": [
+    {
+      "name": "Part 1",
+      "title": "1. 计划基础",
+      "questions": [
+        {
+          "correctAnswer": "是",
```
Comment (goneall) [inline questionnaire-zh-zh.json:22]: correct answer should not be translated

File: questionnaire-zh-zh.json [hunk 14]
```diff
@@ -0,0 +1,434 @@
...
+  "questionsText": "问题",
+  "completedText": "已完成？",
+  "statusText": "状态",
+  "answerColumnText": "回答",
+  "sectionColumnText": "部分",
+  "numberColumnText": "编号",
+  "specRefColumnText": "规范参考",
+  "questionColumnText": "问题文本",
+  "tocText": "目录",
+  "preambleText": "OpenChain项目通过简化开放源许可合规性和保持一致性，建立对于开放源的信任。维护OpenChain规范，定义各品质合规程序必须满足的一组核心需求。其结果是，对于软件供应链参与者来说，开放源许可合规性变得更加可预测、可理解和高效。\n本文档包含一系列确定公司是否符合OpenChain的问题。如果所有问题的答案都为“是”，那么该公司符合OpenChain规范2.1版（功能与ISO\/IEC DIS 5230:2020（e）一致）的所有要求。如果任一问题的答案为“否”，那么公司可以清楚地确定哪些需要追加投资以改进合规程序。",
```
Comment (goneall) [inline questionnaire-zh-zh.json:15]: Not sure about the backslash in the preamble text

File: questionnaire-ja.json [hunk 15]
```diff
@@ -1,434 +1,434 @@
...
+  "questionsText": "質問",
+  "completedText": "終了しましたか？",
+  "statusText": "ステータス",
+  "answerColumnText": "回答",
+  "sectionColumnText": "セクション",
+  "numberColumnText": "番号",
+  "specRefColumnText": "仕様参考",
+  "questionColumnText": "質問本文",
+  "tocText": "目次",
+  "preambleText": "OpenChainプロジェクトは、オープンソースライセンスへのコンプライアンスをよりシンプルにし、一貫性を持たせることで、オープンソースに対する信頼を築きます。OpenChain仕様では、各品質コンプライアンスプログラムが満足すべき核となる要件を定義しています。その結果、オープンソース・ライセンス・コンプライアンスがソフトウェアサプライチェーンの中にいる者にとって、より予測可能でわかりやすく、効率的なものになっています。\nこのドキュメントには企業がOpenChain適合がどうかを判断するための一連の質問が記載されています。すべての質問が「はい」だった場合、その企業はOpenChain仕様2.0版のすべての要件に適合していることになります。いくつかの質問に対する答えが「いいえ」だった場合、その企業はコンプライアンスプロセスの改善のために何が必要かを明確にすることができます。",
```
Comment (goneall) [inline questionnaire-ja.json:15]: Is this missing ISO \/ IEC DIS 5230: 2020 (e)?

File: questionnaire-ko.json [hunk 16]
```diff
@@ -1,434 +1,434 @@
...
+  "questionsText": "질문",
+  "completedText": "완료?",
+  "statusText": "상태",
+  "answerColumnText": "답변",
+  "sectionColumnText": "섹션",
+  "numberColumnText": "No",
+  "specRefColumnText": "설명서 Ref",
+  "questionColumnText": "질문 내용",
+  "tocText": "목차",
+  "preambleText": "OpenChain 프로젝트는 오픈소스 라이선스 컴플라이언스를 보다 간단하고 일관성있게 만들어 오픈소스에 대한 신뢰를 구축합니다. OpenChain 설명서는 모든 우수한 컴플라이언스 프로그램이 충족해야하는 핵심 요건을 정의합니다. 그 결과 소프트웨어 공급망의 참가자가 오픈소스 라이선스 컴플라이언스를 보다 예측 가능하고 이해하기 쉽고 효율적으로 만들 수 있습니다.\n이 문서에는 회사가 OpenChain을 준수하는지 여부를 확인하기위한 일련의 질문이 포함되어 있습니다. 이러한 각 질문에 \"예\"라고 대답 할 수 있으면 해당 회사는 OpenChain 설명서 버전 2.0에 대한 모든 요건을 충족합니다. 질문에 \"아니오\"라고 대답하면 회사는 컴플라이언스 프로세스를 개선하기 위해 추가 투자가 필요한 위치를 명확하게 식별 할 수 있습니다.",
```
Comment (goneall) [inline questionnaire-ko.json:15]: Is this missing ISO \/ IEC DIS 5230: 2020 (e)?

File: questionnaire-ko.json [hunk 17]
```diff
@@ -1,434 +1,434 @@
...
-          "number": "3.a",
-          "type": "YES_NO",
+            "2.1",
+            "2.2.5"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Yes",
```
Comment (goneall) [inline questionnaire-ko.json:238]: This looks incorrect - should be "Part 3"

File: questionnaire-ko.json [hunk 18]
```diff
@@ -1,434 +1,434 @@
...
+          "type": "YES_NO",
           "specReference": [
+            "4.1",
+            "4.1.2"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Yes",
```
Comment (goneall) [inline questionnaire-ko.json:372]: This is a bit messed up - see the following - looks like something got out of alignment during translation

File: questionnaire-ko.json [hunk 19]
```diff
@@ -1,434 +1,434 @@
...
+      "title": "5. 4.c",
+      "questions": [
+        {
+          "correctAnswer": "네",
+          "question": "Part 5",
```
Comment (goneall) [inline questionnaire-ko.json:377]: Same here

File: questionnaire-ko.json [hunk 20]
```diff
@@ -1,434 +1,434 @@
...
           "specReference": [
+            "4.1",
+            "4.1.2"
+          ]
+        }
+      ]
+    },
+    {
+      "name": "Yes",
+      "title": "5. 4.c",
```
Comment (goneall) [inline questionnaire-ko.json:373]: Same here

**Review by goneall — CHANGES_REQUESTED** [review]
Found a few issues that require an update.

Across all the files, the "correctAnswer" should not be translated - you can probably do a quick search and replace to put the English "Yes" back.

Some of the files have translated "name" for the sections - these should not be translated since they are metadata.

A few of the questionnaires look like they got out of sync with section names showing up in questions etc.  I did a search of "name": "Yes" to find them.

There is a backslash in the preamble - it looks like it is trying to escape a forward slash?  I'm not sure if this is correct or if it may trip up the JSON parsers.

File: questionnaire-zh-tw.json [hunk 21]
```diff
@@ -0,0 +1,434 @@
+{
+  "specVersion": "2.1.1",
+  "language": "zh-tw",
```
Comment (goneall) [inline questionnaire-zh-tw.json:3]: Per email conversation - if you could update this to zh-Hant

File: questionnaire-zh-zh.json [hunk 22]
```diff
@@ -0,0 +1,434 @@
+{
+  "specVersion": "2.1.1",
+  "language": "zh-zh",
```
Comment (goneall) [inline questionnaire-zh-zh.json:3]: Per email conversation, if you could change this to zh-Hans

**Comment by brandelune** [comment]
Working on it.

File: questionnaire-zh-zh.json (hunk 22 above)
Comment (brandelune) [reply questionnaire-zh-zh.json]: done locally, sending PR later.

File: questionnaire-zh-tw.json (hunk 21 above)
Comment (brandelune) [reply questionnaire-zh-tw.json]: Done. sending PR later.

File: questionnaire-zh-zh.json (hunk 22 above)
Comment (brandelune) [reply questionnaire-zh-zh.json]: Done. Sending PR later.

File: questionnaire-de.json [hunk 23]
```diff
@@ -0,0 +1,434 @@
...
+          "question": "Haben Sie eine dokumentierte Richtlinie zur Einhaltung der Open Source-Lizenzbestimmungen für die mitgelieferte Software?",
+          "number": "1.a",
+          "type": "YES_NO",
+          "specReference": [
+            "1.1",
+            "1.1.1"
+          ]
+        },
+        {
+          "correctAnswer": "Ja",
```
[the remark this answers is not in the document]
Comment (brandelune) [reply questionnaire-de.json]: Done locally. Pushing later.

File: questionnaire-ja.json (hunk 2 above)
Comment (brandelune) [reply questionnaire-ja.json]: Done locally. Pushing later.

File: questionnaire-ko.json (hunk 3 above)
Comment (brandelune) [reply questionnaire-ko.json]: Done locally. Pushing later.

File: questionnaire-ko.json (hunk 5 above)
Comment (brandelune) [reply questionnaire-ko.json]: Done locally. Pushing later.

File: questionnaire-zh-tw.json (hunk 12 above)
Comment (brandelune) [reply questionnaire-zh-tw.json]: Done locally. Pushing later.

File: questionnaire-zh-zh.json (hunk 13 above)
Comment (brandelune) [reply questionnaire-zh-zh.json]: Done locally. Pushing later.

File: questionnaire-ko.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (brandelune) [reply questionnaire-ko.json:152]: Done here. Pushing later.

File: questionnaire-de.json [hunk 24]
```diff
@@ -0,0 +1,434 @@
...
+  "sectionColumnText": "Abschnitt",
+  "numberColumnText": "Nummer",
+  "specRefColumnText": "Spec Ref",
+  "questionColumnText": "Fragentext",
+  "tocText": "Inhaltsverzeichnis",
+  "preambleText": "Das OpenChain-Projekt schafft Vertrauen in Open Source, indem es die Lizenzkonformität einfacher und konsistenter macht. Es verwaltet die OpenChain-Spezifikation, um einen Kernsatz von Anforderungen zu definieren, die jedes Qualitätskonformitätsprogramm erfüllen muss. Das Ergebnis ist, dass die Einhaltung von Open Source-Lizenzen für die Teilnehmer der Software-Lieferkette vorhersehbarer, verständlicher und effizienter wird.\nDieses Dokument enthält eine Reihe von Fragen, um festzustellen, ob ein Unternehmen über ein OpenChain-Konformitätsprogramm verfügt. Wenn jede dieser Fragen mit \"Ja\" beantwortet werden kann, erfüllt dieses Unternehmen alle Anforderungen an die Konformität mit der OpenChain-Spezifikation Version 2.1 (funktional identisch mit ISO \/ IEC DIS 5230: 2020 (e)). Wenn eine der Fragen mit \"Nein\" beantwortet wird, kann das Unternehmen klar erkennen, wo zusätzliche Investitionen erforderlich sind, um den Compliance-Prozess zu verbessern.",
```
[the remark this answers is not in the document]
Comment (brandelune) [reply questionnaire-de.json]: Done locally, PR later.

File: questionnaire-de.json (hunk 1 above)
Comment (brandelune) [reply questionnaire-de.json]: done locally, PR later.

File: questionnaire-zh-tw.json (hunk 11 above)
Comment (brandelune) [reply questionnaire-zh-tw.json]: done locally, PR later.

File: questionnaire-zh-zh.json (hunk 14 above)
Comment (brandelune) [reply questionnaire-zh-zh.json]: done locally, PR later.

File: questionnaire-ko.json (hunk 18 above)
Comment (brandelune) [reply questionnaire-ko.json]: Indeed. I just fixed the alignment. I should have noticed that before. Sorry. PR later.

File: questionnaire-ko.json (hunk 20 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 17 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 10 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 9 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 8 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 7 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ko.json (hunk 6 above)
Comment (brandelune) [reply questionnaire-ko.json]: PR later.

File: questionnaire-ja.json (hunk 15 above)
Comment (brandelune) [reply questionnaire-ja.json:15]: It looks like the files for JA and KO got mixed up during translation. Fixing now.

File: questionnaire-ko.json (hunk 16 above)
Comment (brandelune) [reply questionnaire-ko.json:15]: It looks like the files for JA and KO got mixed up during translation. Fixing now.

Comment (brandelune) [reply questionnaire-ko.json:15]: I see where the issue comes from. The translators took version 2.1 from the repository as reference and in neither JA nor KO there is the reference to ISO in the "official" translation. I'm fixing JA now at home, asking for urgent fix for KO to the provider.

Comment (brandelune) [reply questionnaire-ko.json]: I just sent a fix. It's included in the PR.

**Comment by brandelune** [comment]
All the issues have been fixed, as far as I can tell. I'll be up at 6:00 am (Tokyo) to check if there are still things to fix before it goes public.

**Comment by brandelune** [comment]
Regarding the travis-ci checked that failed, the message is:

Invalid language zh-Hans  in file ./questionnaire-zh-Hans.json
Invalid language zh-Hant  in file ./questionnaire-zh-Hant.json
npm ERR! Test failed.  See above for more details.

So it looks like the codes are not valid. You may want to check zh-cn/zh-tw as per the message I sent earlier. But maybe it's a different issue, if that's the cas, just ignore.

**Comment by goneall** [comment]
> So it looks like the codes are not valid. You may want to check zh-cn/zh-tw as per the message I sent earlier. But maybe it's a different issue, if that's the case, just ignore.

This turns out to be an issue in the script that validates the codes - turns out there is a TODO in the code to update the pattern.  We can just ignore this for now.

https://github.com/OpenChain-Project/conformance-questionnaire/blob/efbc14ad5a627fc7124f47bc030aaca1d4c8f842/scripts/validate-questionnaire.js#L16

**Review by goneall — APPROVED** [review]
LGTM

**Pull request merged**